src/shared/highlighted-field/use-highlighted-field.js

```diff
@@ -40,26 +46,27 @@ function gatherHighlightedFieldData({ de, coc, dataValueSet }) {
 }
 
 export default function useHighlightedField() {
-    const { data: dataValueSet } = useDataValueSet()
-    const { item } = useHighlightedFieldIdContext()
-    const [currentItem, setHighlightedFieldId] = useState(() => {
-        const { de, coc } = item
-        return gatherHighlightedFieldData({ de, coc, dataValueSet })
-    })
+    const item = useHighlightedFieldStore((state) => state.item)
+    const dataValue = useValueStore((state) =>
+        state.getDataValue({
+            dataElementId: item?.dataElementId,
+            categoryOptionComboId: item?.categoryOptionComboId,
+        })
+    )
+    const { data: metadata } = useMetadata()
+    const dataElement = selectors.getDataElementById(
+        metadata,
+        item?.dataElementId
+    )
```

Comment:
I thought the way the `datElement` was attached to the `item` in the context was quite confusing. So I find it's cleaner to just store the ID, and then we grab the dataElement here.